### PR TITLE
Add KiCad flatpak

### PR DESCRIPTION
**Pull request by lkundrak**
My first Flatpak and it's not like I'd know what I am really doing yet.
(This probably deserves a careful review.)

### CODE REVIEW

File: org.kicad_pcb.KiCad.json [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        "--socket=x11",
+        "--filesystem=home"
+    ],
+    "cleanup": ["/share/doc",
+        "/share/aclocal", "/share/bakefile",
+        "/lib/wx", "/include",
+        "*.la", "*.a"],
+    "modules": [
+        {
+            "name": "libglu",
```
Comment (nedrichards) [inline org.kicad_pcb.KiCad.json]: Use the [shared module](https://github.com/flathub/shared-modules) here.

File: org.kicad_pcb.KiCad.json [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://mesa.freedesktop.org/archive/glu/glu-9.0.0.tar.bz2",
+                    "sha256": "1f7ad0d379a722fcbd303aa5650c6d7d5544fde83196b42a73d1193568a4df12"
+                }
+            ]
+        },
+        {
+            "name": "GLEW",
```
Comment (nedrichards) [inline org.kicad_pcb.KiCad.json]: Oh interesting, the shared module has 2.0 here, might be worth testing and updating @TingPing @Barthalion

File: org.kicad_pcb.KiCad.json [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://downloads.sourceforge.net/project/glew/glew/2.1.0/glew-2.1.0.tgz",
+                    "sha256": "04de91e7e6763039bc11940095cd9c7f880baba82196a7765f727ac05a993c95"
+                }
+            ]
+        },
+        {
+            "name": "wxWidgets",
```
Comment (nedrichards) [inline org.kicad_pcb.KiCad.json]: experience here is that you're going to need to force a reconfigure to make this understand Aarch64. See examples [in flathub](https://github.com/search?q=org%3Aflathub+wxwidgets&type=Code).

**Comment by nedrichards** [comment]
Test build https://flathub.org/builds/#/builders/1/builds/777

**Comment by alexlarsson** [comment]
bot, build org.kicad_pcb.KiCad

**Comment by nedrichards** [comment]
Yep, aside from the aarch64 stuff mentioned above and the other review comments, looks like it builds fine.

File: org.kicad_pcb.KiCad.json (hunk 2 above)
Comment (Barthalion) [reply org.kicad_pcb.KiCad.json]: Done.

**Comment by lkundrak** [comment]
bot, build org.kicad_pcb.KiCad

**Comment by lkundrak** [comment]
It won't listen to me, will it? :)

**Comment by nedrichards** [comment]
Sadly not! It will listen to you inside the repo once it's accepted and added to flathub though, so you can test PRs.

**Comment by nedrichards** [comment]
bot, build org.kicad_pcb.KiCad

**Comment by lkundrak** [comment]
This is basically a half-blind shot at fixing the 64-bit ARM build. I started a build on a borrowed box, but it is taking longer than I am able to wait (though the boost context library did).

Perhaps it would be better idea to have buildbot build it.

**Comment by Barthalion** [comment]
bot, build org.kicad_pcb.KiCad

**Comment by lkundrak** [comment]
Uh. Let's fix the aarch64 build differently.

**Comment by lkundrak** [comment]
(Verified to build well locally)

**Comment by nedrichards** [comment]
bot, build org.kicad_pcb.KiCad

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.kicad_pcb.KiCad

**Pull request closed without merging**